### PR TITLE
Add widget and dashboard code

### PR DESCRIPTION
**Pull request by rihorn2**
still rough, but in the process, and the structure is all there

### CODE REVIEW

File: fairlearn/widget/FairlearnWidget.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+"""Defines the python side of the shared state of the Fairlearn widget."""
```
Comment (romanlutz) [inline fairlearn/widget/FairlearnWidget.py]: fairlearn all lower case please here and in every file, excluding class names, of course

File: fairlearn/widget/js/LICENSE [hunk 2]
```diff
@@ -0,0 +1,15 @@
+This Preview is made available to you on the condition that you agree to the
+[Supplemental Terms of Use for Microsoft Azure Previews][1], which supplement
```
Comment (romanlutz) [inline fairlearn/widget/js/LICENSE]: Azure? I don't think we want such things there?

File: fairlearn/widget/js/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+Machine Learning fairness widget for Jupyter widget
+
+Package Install
+---------------
+
+**Prerequisites**
+- [node](http://nodejs.org/)
+
+```bash
+npm install --save fairnlearn-widget
```
Comment (romanlutz) [inline fairlearn/widget/js/README.md]: typo fairnlearn

File: fairlearn/widget/js/package.json [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  "name": "fairlearn-widget",
+  "version": "0.1.0",
+  "description": "Fairlearn widget for Jupyter notebooks",
+  "author": "Brandon Horn (rihorn@microsoft.com)",
+  "license": "See license file 'LICENSE'",
+  "main": "lib/index.js",
+  "types": "./lib/index.d.ts",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/microsoft/fairlearn"
```
Comment (romanlutz) [inline fairlearn/widget/js/package.json]: adjust url please

File: visualization/FairnessDashboard/src/Controls/ParityPicker.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
+import { IParityPickerProps } from "../FairnessWizard";
```
Comment (romanlutz) [inline visualization/FairnessDashboard/src/Controls/ParityPicker.tsx]: do these files need copyright?

File: fairlearn/widget/FairlearnDashboard.py [hunk 6]
```diff
@@ -0,0 +1,122 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+"""Defines the Fairlearn dashboard class."""
+
+from .FairlearnWidget import FairlearnWidget
+from IPython.display import display
```
Comment (romanlutz) [inline fairlearn/widget/FairlearnDashboard.py]: are there any new dependencies we need to add now that we don't have a separate package anymore? I mean dependencies that we'd need to add to setup.py & requirements.txt

File: fairlearn/widget/FairlearnWidget.py (hunk 1 above)
Comment (rihorn2) [reply fairlearn/widget/FairlearnWidget.py]: will do. Question though, the design has "FairLearn" in the title displayed of the widget in several places. Is the plan fairlearn in all code, but its ok in presentation?

File: visualization/FairnessDashboard/src/Controls/ParityPicker.tsx (hunk 5 above)
Comment (rihorn2) [reply visualization/FairnessDashboard/src/Controls/ParityPicker.tsx]: hmm, no one has ever mentioned it?

File: fairlearn/widget/FairlearnDashboard.py (hunk 6 above)
Comment (rihorn2) [reply fairlearn/widget/FairlearnDashboard.py]: the setup.py has been updated to include all the widget pieces in the python wheel (I hope)

**Pull request merged**